weave-js/src/components/PagePanelComponents/Home/Browse3/compare/refUtil.ts

```diff
@@ -0,0 +1,28 @@
+import {isWeaveRef} from '../filters/common';
+import {traverse, TraverseContext} from '../pages/CallPage/traverse';
+import {isExpandableRef} from '../pages/wfReactInterface/tsDataModelHooksCallRefExpansion';
+import {ComparableObjects} from './types';
+
+// When we replace a ref with its resolved data we insert this key
+// with the ref URI to keep track of the original value.
+export const RESOLVED_REF_KEY = '_ref';
```

Comment:
I think we do something similar for call expansion - maybe an opportunity for some merging.

Reply:
Yeah, I think there's a lot of opportunity for merging functionality, though with all of the problems we have with things like import cycle I'm a little wary of introducing dependencies across different parts of the code and willing to live with more duplication than I'd prefer.